protocol/messenger_backup_handler.go

```diff
@@ -0,0 +1,205 @@
+package protocol
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts/settings"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/wakusync"
+)
+
+const (
+	// timeToPostponeBackedUpMessagesHandling - the idea is to wait for 30 secs in loading state with the hope that at least
+	// one message from the set of last backed up messages will arrive, that we can know which set of messages to work with
+	timeToPostponeBackedUpMessagesHandling = 30 * time.Second
+
+	SyncWakuSectionKeyProfile     = "profile"
+	SyncWakuSectionKeyContacts    = "contacts"
+	SyncWakuSectionKeyCommunities = "communities"
+	SyncWakuSectionKeySettings    = "settings"
+)
+
+type backupHandler struct {
+	messagesToProceed      []protobuf.Backup
+	lastKnownTime          uint64
+	postponeHandling       bool
+	postponeTasksWaitGroup sync.WaitGroup
+}
+
+func (bh *backupHandler) addMessage(message protobuf.Backup) {
+	if message.Clock < bh.lastKnownTime {
+		return
+	}
+	if message.Clock > bh.lastKnownTime {
+		bh.messagesToProceed = nil
+		bh.lastKnownTime = message.Clock
+	}
+
+	bh.messagesToProceed = append(bh.messagesToProceed, message)
+}
+
+func (m *Messenger) startWaitingForTheLatestBackedupMessageLoop() {
+	ticker := time.NewTicker(timeToPostponeBackedUpMessagesHandling)
+	m.backedupMessagesHandler.postponeTasksWaitGroup.Add(1)
+	go func() {
+		defer m.backedupMessagesHandler.postponeTasksWaitGroup.Done()
```

Comment:
`ticker` should probably be part of the `go func()`

Reply:
Also, probably makes sense to do something like:
```go
ticker := time.NewTicker(timeToPostponeBackedUpMessagesHandling)
defer ticker.Stop()
```

because inside the case of <-ticker.C there is a `return`, meaning that the ticker would be left dangling since it's only stopped on `<-m.quit`

Reply:
Good point, I have to call `ticker.Stop()` before return definitely, cause according to documentation `Stop the ticker to release associated resources`, so need to update it like this:
```
if !m.backedupMessagesHandler.postponeHandling {
        ticker.Stop() // <--need to add this
	return
}
```